pyat/at/physics/energy_loss.py

```diff
@@ -104,16 +160,13 @@ def set_cavity_phase(ring, method=ELossMethod.INTEGRAL,
     if cavpts is None and refpts is not None:
         warn(FutureWarning('You should use "cavpts" instead of "refpts"'))
         cavpts = refpts
-    rfv = ring.get_rf_voltage(cavpts=cavpts)
-    freq = ring.get_rf_frequency(cavpts=cavpts)
+    elif cavpts is None:
+        cavpts = get_refpts(ring, RFCavity)
```

Comment:
Redundant test: caveats if properly handled in both `get_timelag_fromU0` and `set_value_refpts`

Reply:
set_value_refpts does not handle refpts=None, I am not I understand what you mean